iot/api-client/manager periodic testing fails

```
------------------------------------------------------------
- testing iot/api-client/manager
------------------------------------------------------------

[ERROR] Tests run: 21, Failures: 0, Errors: 17, Skipped: 4, Time elapsed: 494.087 s <<< FAILURE! - in com.example.cloud.iot.examples.ManagerIT
[ERROR] com.example.cloud.iot.examples.ManagerIT.testMqttDeviceState  Time elapsed: 486.849 s  <<< ERROR!
org.eclipse.paho.client.mqttv3.MqttException: Client is not connected
	at com.example.cloud.iot.examples.ManagerIT.testMqttDeviceState(ManagerIT.java:589)

[ERROR] com.example.cloud.iot.examples.ManagerIT.testHttpDeviceConfig  Time elapsed: 0.851 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testMqttDeviceConfig  Time elapsed: 0.374 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testMqttDeviceEvents  Time elapsed: 0.364 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateGetDevice  Time elapsed: 0.52 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testPatchEs  Time elapsed: 0.361 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateConfigureDevice  Time elapsed: 0.537 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateGetRegistry  Time elapsed: 0.434 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateListDevices  Time elapsed: 0.318 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateDeleteEsDevice  Time elapsed: 0.53 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testHttpDeviceEvent  Time elapsed: 0.359 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testHttpDeviceState  Time elapsed: 0.527 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testGetIam  Time elapsed: 0.668 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateDeleteRsaDevice  Time elapsed: 0.195 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateDeleteUnauthDevice  Time elapsed: 0.333 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testSetIam  Time elapsed: 0.368 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testPatchRsa  Time elapsed: 0.345 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-15882783469).

[ERROR] Errors: 
[ERROR]   ManagerIT.testCreateConfigureDevice » AlreadyExists io.grpc.StatusRuntimeExcep...
[ERROR]   ManagerIT.testCreateDeleteEsDevice » AlreadyExists io.grpc.StatusRuntimeExcept...
[ERROR]   ManagerIT.testCreateDeleteRsaDevice » AlreadyExists io.grpc.StatusRuntimeExcep...
[ERROR]   ManagerIT.testCreateDeleteUnauthDevice » AlreadyExists io.grpc.StatusRuntimeEx...
[ERROR]   ManagerIT.testCreateGetDevice » AlreadyExists io.grpc.StatusRuntimeException: ...
[ERROR]   ManagerIT.testCreateGetRegistry » AlreadyExists io.grpc.StatusRuntimeException...
[ERROR]   ManagerIT.testCreateListDevices » AlreadyExists io.grpc.StatusRuntimeException...
[ERROR]   ManagerIT.testGetIam » AlreadyExists io.grpc.StatusRuntimeException: ALREADY_E...
[ERROR]   ManagerIT.testHttpDeviceConfig » AlreadyExists io.grpc.StatusRuntimeException:...
[ERROR]   ManagerIT.testHttpDeviceEvent » AlreadyExists io.grpc.StatusRuntimeException: ...
[ERROR]   ManagerIT.testHttpDeviceState » AlreadyExists io.grpc.StatusRuntimeException: ...
[ERROR]   ManagerIT.testMqttDeviceConfig » AlreadyExists io.grpc.StatusRuntimeException:...
[ERROR]   ManagerIT.testMqttDeviceEvents » AlreadyExists io.grpc.StatusRuntimeException:...
[ERROR]   ManagerIT.testMqttDeviceState:589 » Mqtt Client is not connected
[ERROR]   ManagerIT.testPatchEs » AlreadyExists io.grpc.StatusRuntimeException: ALREADY_...
[ERROR]   ManagerIT.testPatchRsa » AlreadyExists io.grpc.StatusRuntimeException: ALREADY...
[ERROR]   ManagerIT.testSetIam » AlreadyExists io.grpc.StatusRuntimeException: ALREADY_E...
[ERROR] Tests run: 21, Failures: 0, Errors: 17, Skipped: 4
```

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -115,6 +115,16 @@ public class ManagerIT {
             DeviceRegistryExample.clearRegistry(CLOUD_REGION, PROJECT_ID, registryId);
           }
         }
+        // Also remove the current test registry
+        try {
+          DeviceRegistryExample.clearRegistry(CLOUD_REGION, PROJECT_ID, REGISTRY_ID);
+        } catch (com.google.api.client.googleapis.json.GoogleJsonResponseException gjre) {
+          if (gjre.getStatusCode() == 404) {
+            // Expected, the registry resource is available for creation.
+          } else {
+            throw gjre;
+          }
+        }
       }
     } else {
       System.out.println("Project has no registries.");

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -55,7 +55,7 @@ public class ManagerIT {
   private static final String CLOUD_REGION = "us-central1";
   private static final String ES_PATH = "resources/ec_public.pem";
   private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
-  private static final String REGISTRY_ID = "java-reg" + (System.currentTimeMillis() / 100L);
+  private static final String REGISTRY_ID = "java-reg-" + (System.currentTimeMillis() / 100L);
   private static final String RSA_PATH = "resources/rsa_cert.pem";
   private static final String PKCS_PATH = "resources/rsa_private_pkcs8";
   private static final String TOPIC_ID = "java-pst-" + (System.currentTimeMillis() / 100L);

